### DISCUSSION TITLE
Do I need to have QGIS with conda installed?

### DISCUSSION THREAD

**Discussion by ambarja** [geojupyter/jupytergis#215] [Q&A] [answered]
Hello , this is a very interesting and fantastic project.
My question is about how to open qgis projects with JupyterGIS, is it necessary to have QGIS installed with conda ? 
Most of the users have QGIS Desktop installed in the traditional way, so is it possible to configure the PATH maybe from this QGIS already installed and JupyterGIS can recognize it?

**Comment by martinRenou** [comment, marked as answer]
Hello thanks for the kind words! 

Yes, as soon as the QGIS Python API is importable, that should work.

Are you using the jupytergis conda package ? It does have a dependency on QGIS from conda but we may want to make that optional.

Alternatively you can install jupytergis with pip for now and that will not pull QGIS.

**Reply by mfisher87** [reply]
@ambarja Hope you're doing well! Did Martin's response above resolve your issue? If so, can you please click "Mark as answer"? Thanks! :bow:

**Reply by b-a0** [reply]
It's good to see that it works as "soon as the QGIS Python API is importable", but how can one make it importable? The original question has:

>  ...is it possible to configure the PATH maybe from this QGIS already installed and JupyterGIS can recognize it?

So, is it possible ot set the PATH used by JupyterGIS?

**Reply by mfisher87** [reply]
We have considered an option to enable importing from the system-installed QGIS, and there is a PR open here: https://github.com/geojupyter/jupytergis/pull/1064

There are some risks to this approach, including that the user's system-installed QGIS is incompatible with JupyterGIS. Please check out the PR for more on this. @lbartoletti provided a pathway to use system-installed QGIS with an environment variable to allow users to opt in to this behavior at their own risk. @b-a0 Would you use such a feature, and would you be comfortable with us being unable to offer any guarantees that it would work as you expect?

Using QGIS installed via conda-forge is always going to be our recommended approach.

Related: https://github.com/conda-forge/jupytergis-packages-feedstock/issues/78

**Reply by b-a0** [reply]
Thanks for linking that PR, I only searched prior issues and therefore did not find that.

> Would you use such a feature

Yes, would definitely try it out as I want to move away from conda/mamba to uv as conda feels very slow (both in creating envs/installing packages as well as in using it in e.g. VS code)

> would you be comfortable with us being unable to offer any guarantees

Of course, I can always fallback on the conda approach if it doesn't work, and with open source projects there's no guarantee anyway :)
